tensorflow/python/ops/math_ops.py

```diff
@@ -4857,7 +4858,207 @@ def sparse_segment_sum(
+      multiplied. Must have rank > 1.
+    values: `tf.Tensor` containing the values to be scaled and added to the
+      sampled dot product.
+    dense_shape: `tf.Tensor` defining the dense shape of the output.
+    mat1: `tf.Tensor` to be multiplied. Must have rank > 1.
+    mat2: `tf.Tensor` to be multiplied. Must have rank > 1.
+    beta: Number to be multipled with `values`. Defaults to 1.0.
+    alpha: Number to be multiplied with the sampled dot product of `mat1` and
+      `mat2`. Defaults to 1.0.
+    output_type: the output datatype if needed. Defaults to float32.
+
+  Returns:
+    A tuple of three `tf.Tensor` objects (indices, result, dense_shape) making
+    up the components of a `SparseTensor` representing the result of the
+    operation.
+
+    result = `alpha` * (`mat1` @ `mat2`) * spy(`indices`) + `beta` * `values`
+
+    The function `spy(indices)` is the sparsity pattern matrix derived from
+    `indices`. 
+  """
+  indices = ops.convert_to_tensor(indices)
+  values = ops.convert_to_tensor(values, dtype=output_type)
+  dense_shape = ops.convert_to_tensor(dense_shape, dtype=dtypes.int32)
+  mat1 = ops.convert_to_tensor(mat1, dtype=output_type)
+  mat2 = ops.convert_to_tensor(mat2, dtype=output_type)
+
+  mat1_shape = mat1.shape
+  mat2_shape = mat2.shape
+  
+  if mat1_shape is None:
+    mat1_shape = array_ops.shape(mat1)
+  if mat2_shape is None:
+    mat2_shape = array_ops.shape(mat2)
+
+  dense_rows = mat1_shape[-2]
+  dense_cols = mat2_shape[-1]
+
+  output_shape = constant_op.constant([dense_rows, dense_cols], 
+                                      dtype=dtypes.int32)
+
+  condition = reduce_all(equal(dense_shape, output_shape))
+
+  # Use dense_shape to validate matrix shapes
+  if context.executing_eagerly():
+    if not condition:
+      raise ValueError(
+              f"Dense shape: {dense_shape} does not match "
+              f"output shape: {output_shape}")
+  else: # not context.executing_eagerly()
+    dense_shape_static = tensor_util.constant_value(dense_shape)
+    output_shape_static = tensor_util.constant_value(output_shape)
+    if dense_shape_static is not None and output_shape_static is not None:
+      condition_static = np.all(np.equal(dense_shape_static, 
+                                         output_shape_static)) 
+      if not condition_static:
+        raise ValueError(
+                f"Dense shape: {dense_shape} does not match "
+                f"output shape: {output_shape}")
+    
+    data = [
+        'Dense shape: ', 
+        dense_shape, 
+        ' does not match output shape: ', 
+        output_shape
+    ]
+ 
+    gen_logging_ops._assert(condition, data, None, name="Assert")
+
+  # Extract row and column indices
+  batch_indices = indices[..., :-2]
+  row_indices = indices[..., :-1]
+  col_indices = array_ops.concat([batch_indices, 
+                                  indices[..., -1:]], axis=-1)
+
+  # Calculate batch dimensions
+  rank = mat1.ndim
```

Comment:
Use `array_ops.rank(mat1)` here directly

Reply:
Changing this causes my tests to fail as the `batch_dims` parameter on `array_ops.gather_nd()` expects an `int32` but gets a symbolic tensor